### PR TITLE
fix: Icon Selector width on propety pane resize

### LINKED ISSUE #11735: [Bug]: Icon Selector width is not updated when we resize the property pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/10436935/157429002-608d789e-aa4c-4338-803c-cbd86f11eaf0.mov

### Steps To Reproduce

1. Add a ButtonWidget
2. Open the icon selector in property pane
3. Try to resize the property pane
4. open the icon selector again and see the width of the icon selector

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Icon Selector's width was not recalculated when the property pane was resized.
This also fixes an issue in JSONForm Widget where it is unable to display Icon Selector if the property pane section is hidden initially.

Thanks, @ashit-rath for coming up with a good solution.

Fixes #11735 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/icon-selector-width 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.87 **(-0.01)** | 37.25 **(0.02)** | 35.23 **(-0.01)** | 56.2 **(-0.01)**
 :green_circle: | app/client/src/components/propertyControls/IconSelectControl.tsx | 91.3 **(-0.48)** | 76.11 **(3.21)** | 95.83 **(-0.32)** | 91.34 **(-0.33)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/components/propertyControls/IconSelectControl.tsx [hunk 1]
```diff
@@ -177,10 +161,13 @@ class IconSelectControl extends BaseControl<
 
   public render() {
     const { defaultIconName, propertyValue: iconName } = this.props;
-    const { activeIcon, popoverTargetWidth } = this.state;
+    const { activeIcon } = this.state;
+    const containerWidth =
+      this.iconSelectTargetRef.current?.getBoundingClientRect().width || 0;
```
[the remark this answers is not in the document]
Comment (aswathkk) [reply app/client/src/components/propertyControls/IconSelectControl.tsx:166]: I didn't realise optional chaining works for function calls as well. Nice info

**Review by ashit-rath — APPROVED** [review]

**Comment by ashit-rath** [comment]
@aswathkk Is this ready to merge?

**Comment by aswathkk** [comment]
Yes @ashit-rath.
Waiting for approval from @vivekverma2312

**Comment by vivekverma2312** [comment]
@aswathkk Tested the issue and it's working fine.

**Pull request merged**